### ISSUE TITLE
Bug in MinimalFaithfulPermutationDegree

### ISSUE THREAD

**Issue by hungaborhorvath** [gap-system/gap#3636]
Not sure if it's related to https://mail.gap-system.org/pipermail/forum/2019/005947.html but I report it anyhow:
````
 ┌───────┐   GAP 4.10dev-2247-g6e34384 of today
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-default64-kv7
 Configuration:  gmp 6.1.2, GASMAN, readline
 Loading the library and packages ...
 Packages:   AClib 1.3.1, Alnuth 3.1.1, AtlasRep 2.1.0, AutoDoc 2019.07.24, 
             AutPGrp 1.10.1, Browse 1.8.8, Carat 2.2.3, CRISP 1.4.4, 
             Cryst 4.1.19, CrystCat 1.1.9, CTblLib 1.2.2, FactInt 1.6.2, 
             FGA 1.4.0, Forms 1.2.5, GAPDoc 1.6.3, genss 1.6.6, IO 4.7.0, 
             IRREDSOL 1.4, LAGUNA 3.9.3, orb 4.8.2, Polenta 1.3.8, 
             Polycyclic 2.14, PrimGrp 3.3.2, RadiRoot 2.8, recog 1.3.2, 
             ResClasses 4.7.2, SmallGrp 1.3, Sophus 1.24, SpinSym 1.5.1, 
             TomLib 1.2.8, TransGrp 2.0.4, utils 0.65
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
gap> G := SmallGroup(36,6); L:=DirectFactorsOfGroup(G);
<pc group of size 36 with 4 generators>
[ Group([ f2 ]), Group([ f4, f1 ]) ]
gap> MinimalFaithfulPermutationDegree(G);
10
gap> MinimalFaithfulPermutationDegree(L[2]);
12
gap> 
````
I would expect the subgroup having smaller or equal ````MinimalFaithfulPermutationDegree````.

A minimal working example is the dihedral group of the 6-gon (````D12````), which GAP thinks can be represented in S_5, even though it has a cyclic subgroup of 6 elements for which it thinks that the minimal degree is 6:
````
gap> MinimalFaithfulPermutationDegree(DihedralGroup(12));
5
gap> MinimalFaithfulPermutationDegree(CyclicGroup(6));
6
gap> 
````
Now, the cyclic group on 6 elements can b erepresented in S_5. In fact, for abelian groups in general the minimal degree is the sum of minimal degrees of its cyclic direct factors. 

Problem affects current stable GAP 4.10.2 of 19-Jun-2019, as well.

Edit: the working example is expanded to make clear what is wrong.

**Comment by hungaborhorvath** [comment]
I took a quick look, it seems that ````lib/grplatt.gi:3270```` is invoked. Is it possible that the code assumes transitive permutation representations when the list ```deg```` is created?

**Comment by Stefan-Kohl** [comment]
The bug even shows up when you give the cyclic group of order 6 as a subgroup of S_5:
```
    gap> MinimalFaithfulPermutationDegree(Group((1,2)(3,4,5)));
    6
```

**Comment by fingolfin** [comment]
ping @hulpke who wrote this code

**Comment by hulpke** [comment]
Yes, I have a fix.

**Comment by hulpke** [comment]
@hungaborhorvath No, the code does not assume transitive, but the way the logic of the `if' statement is phrased, the normal case needs to be checked first, before the possible shortcut on subgroups.

**Comment by alex-konovalov** [comment]
@hulpke I am going to branch stable-4.11 today. I've assigned this issue to GAP 4.11 milestone to remind us to backport the fix to stable-4.11  branch when it will be committed to the master branch.

**Comment by fingolfin** [comment]
@hulpke could you submit your fix as a PR, please?
